### PR TITLE
More padding configurations - left/right, duration/num_frames/num_samples

### PR DESCRIPTION
**Pull request by pzelasko**
@danpovey this is going to enable adding a specific number of padding frames in left and right context for ASR training (#189).

### CODE REVIEW

File: lhotse/cut.py [hunk 1]
```diff
@@ -422,34 +423,38 @@ def truncate(
...
+        or a specific number of samples `num_samples`. The three arguments are mutually exclusive.
 
+        :param cut: Cut to be padded.
         :param duration: The cut's minimal duration after padding.
+        :param num_frames: The cut's total number of frames after padding.
+        :param num_samples: The cut's total number of samples after padding.
         :param pad_feat_value: A float value that's used for padding the features.
             By default we assume a log-energy floor of approx. -23 (1e-10 after exp).
-        :return: a padded MixedCut if ``duration`` is greater than this cut's duration, otherwise ``self``.
+        :param direction: string, 'left' or 'right'. Determines whether the padding is added before or after
```
Comment (danpovey) [inline lhotse/cut.py:446]: Perhaps it would be useful to support 'both', or is there a reason why this isn't a good idea?

Comment (danpovey) [reply lhotse/cut.py:446]: .. 'both' could just recurse to the 'left' and 'right' forms if convenient.

Comment (pzelasko) [reply lhotse/cut.py:446]: makes sense, will add

File: lhotse/dataset/transforms/extra_padding.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from lhotse import CutSet
+from lhotse.utils import LOG_EPSILON
+
+
+class ExtraPadding:
```
Comment (pzelasko) [inline lhotse/dataset/transforms/extra_padding.py:5]: @danpovey we can use this one in snowfall (and when we move on to long-recording corpora we can extend it to find and use the real acoustic context)

**Comment by pzelasko** [comment]
I merged since the tests are OK, I can address any comments in a separate PR.

**Pull request merged**